### ISSUE TITLE
RCM API

### ISSUE THREAD

**Issue by teg** [osbuild/osbuild-composer#192]
Introduce a limited API for RCM to use to push composes to osbuild-composer.

Acceptance criteria:

- [x] Introduce a dedicated TCP socket for this API.
- [x] Implement a method to push a new compose, specifying the architectures to build for and the repositories to use. It would be ok to silently drop requests for unsupported (non-native) architecture, until we have support for more. For the time being assume the qcow2 output type and the local upload target.
- [x] Implement a method to get the status of a compose (waiting/running/finished/failed)
- [x] Make the new API optional (ship the systemd socket unit in a sub-package and gracefully handle when not installed, possibly needing further discussion?)
- [ ] Add support for the koji upload target, and use that exclusively in stead of the local one. This requires changes both to the input to the endpoint pushing a new compose, and the output of the status endpoint in finished state.
- [ ] Return per-job status from GET /v1/compose/

Optional:
- [ ] Support kerberos authentication in the same way brew etc does.
- [ ] Extend the status API to optionally block until the compose is no longer waiting/running.

**Comment by msehnout** [comment]
Additional changes:
 - [ ] use systemd named sockets
 - [ ] refactor the store to support multiple distros and arches

**Comment by msehnout** [comment]
Status update:
 * Socket unit is on the way: #250 
 * Upload target is blocked by the fact that the Koji support is missing
 * Kerberos support seems to be low-priority compared to the effort it would take (we can implement TLS support if there is a need for access management)

**Comment by teg** [comment]
This has been superseded by other work.
